service/lib/agama/storage/configs/search.rb

```diff
@@ -37,6 +37,10 @@ class Search
         # @return [:create, :skip, :error]
         attr_accessor :if_not_found
 
+        # Optional max number of devices to match
+        # return [Integer, nil]
+        attr_accessor :max
```

Comment:
NP: I would explicitly mention that nil means it finds all unassigned devices that match the condition.

Reply:
Done via `push -f`